pravega/src/main/scala/akka/stream/alpakka/pravega/settings.scala

```diff
@@ -0,0 +1,246 @@
+                            val timeout: Long) {
+
+  def withGroupName(name: String) = copy(groupName = name)
+  def withReaderId(id: String) = copy(readerId = id)
+  def withTimeout(t: Long) = copy(timeout = t)
+
+  def withSerializer[A](serializer: Serializer[A]) =
+    new ReaderSettings[A](this, serializer)
+
+  def copy(clientConfig: ClientConfig = clientConfig,
+           readerConfig: ReaderConfig = readerConfig,
+           groupName: String = groupName,
+           readerId: String = readerId,
+           timeout: Long = timeout) =
+    new ReaderSettingsBuilder(clientConfig, readerConfig, groupName, readerId, timeout)
+}
+
+class ReaderSettings[A](val settings: ReaderSettingsBuilder, val serializer: Serializer[A] = null) {
+  def groupName = settings.groupName
+  def readerId = settings.readerId
+  def readerConfig = settings.readerConfig
+  def clientConfig = settings.clientConfig
+  def timeout = settings.timeout
+}
+
+object ReaderSettingsBuilder {
+  val configPath = "akka.alpakka.pravega"
+
+  /**
+   * Create reader settings from the default configuration
+   * `akka.alpakka.pravega`.
+   */
+  def apply[A](actorSystem: ActorSystem): ReaderSettingsBuilder =
+    apply(actorSystem.settings.config.getConfig(configPath))
+
+  /**
+   * Create reader settings from the default configuration
+   * `akka.alpakka.pravega`.
+   * @param clientConfigBuilder customisation function
+   */
+  def apply[A](actorSystem: ActorSystem, clientConfigBuilder: ClientConfigBuilder => Unit): ReaderSettingsBuilder =
+    apply(actorSystem.settings.config.getConfig(configPath), Some(clientConfigBuilder))
+
+  /**
+   * Java API: Create reader settings from the default configuration
+   * `akka.alpakka.pravega`.
+   */
+  def create[A](actorSystem: ActorSystem): ReaderSettingsBuilder =
+    apply(actorSystem.settings.config.getConfig(configPath))
+
+  /**
+   * Java API: Create reader settings from the default configuration
+   * `akka.alpakka.pravega`.
+   * @param clientConfigBuilder customisation function
+   */
+  def create[A](actorSystem: ActorSystem, clientConfigBuilder: Consumer[ClientConfigBuilder]): ReaderSettingsBuilder =
+    apply(actorSystem.settings.config.getConfig(configPath), Some(clientConfigBuilder.asScala))
+
+  /**
+   * Create settings from a configuration with the same layout as
+   * the default configuration `akka.alpakka.pravega.reader`.
+   */
+  def apply[A](config: Config,
+               clientConfigBuilder: Option[ClientConfigBuilder => Unit] = None): ReaderSettingsBuilder = {
+
+    import ConfigHelper._
+
+    implicit val rConfig = config.getConfig("reader")
+
+    val readerBasicSetting = new ReaderBasicSetting()
+
+    extractString("group-name")(readerBasicSetting.withGroupName)
+    extractString("reader-id")(readerBasicSetting.withReaderId)
+    extractLong("timeout")(readerBasicSetting.withTimeout)
+
+    val clientConfig = ConfigHelper.buildClientConfig(config, "client-config", clientConfigBuilder)
```

Comment:
With this implementation the settings look like this:

```
akka.alpakka.pravega{
  client-config {
  }
  reader {
  }
  writer {
  }
}
```

And all reader and writer instances share a unique `client-config`. A low-cost improvement would be something like:

```
akka.alpakka.pravega {
  defaults.client-config {
  }
  reader {
	client-config = ${akka.alpakka.pravega.defaults.client-config}
  }
  writer {
	client-config = ${akka.alpakka.pravega.defaults.client-config}
  }
}
```

But we can probably do that in a separate PR.

Reply:
Excellent, thanks you so much for the trick !